src/utils/DataProcessingUtils.js

```diff
@@ -296,94 +331,107 @@ function processEntityValueMap(
  * the "page1_section1" keys are ignored. the "__@@__" keys are the hacky strings that hold hints for correctly
  * processing the entity data.
  */
-function processEntityData(data :Map, edm :Map, mappers :Map = Map()) :{} {
-
-  if (!Immutable.isMap(data) || data.isEmpty()) {
-    const errorMsg :string = '"data" param must be a non-empty immutable map';
-    LOG.error(errorMsg, data);
-    throw new Error(errorMsg);
-  }
-
-  if (!Immutable.isMap(edm) || edm.isEmpty()) {
-    const errorMsg :string = '"edm" param must be a non-empty immutable map';
-    LOG.error(errorMsg, edm);
-    throw new Error(errorMsg);
-  }
-
-  if (!Immutable.isMap(mappers)) {
-    const errorMsg :string = '"mappers" param must be an immutable map';
-    LOG.error(errorMsg, mappers);
-    throw new Error(errorMsg);
-  }
-
-  let processedData :Map = Map();
-
-  data
-    .valueSeq()
+function processEntityData(
+  data :Object | Map,
+  entitySetIds :Object | Map, // { <entitySetName>: <UUID> }
+  propertyTypeIds :Object | Map, // { <propertyTypeFQN>: <UUID> }
+  mappers :Object | Map = {}
+) :Object {
+
+  // if (!Immutable.isMap(data) || data.isEmpty()) {
```

Comment:
Why?

Reply:
I took these out initially because I was trying to support both Immutable collections and plain JS but totes forgot to add these checks back in.